### DISCUSSION TITLE
Research log 2024-09-14 - Proving out the general approach and theory

### DISCUSSION THREAD

**Discussion by daveshap** [daveshap/Raspberry#15] [Announcements]
Today I recorded the first video and verified a few basic things, basically proving out the possibility of using a tool like Claude to not only synthesize reasoning data, but verify it's accurate, and clean it up. I have reaffirmed the general approach.

1. Synthesize a boatload of good questions. Specifically many questions across numerous domains, which require complex reasoning, math, etc. 
2. Synthesize a bunch of conversations (solutions to those questions). We will likely need a "user proxy" agent to encourage the model along.
3. Verify the integrity and quality of the solution. Claude was able to do that on a simple data sample. However, this could get tricky. 
4. Clean up the data for a finetune run. Claude was able to reformat the data, but will take some finesse on the prompting to maintain the reasoning steps. 

All commits recorded here: https://github.com/daveshap/raspberry_experiments/commit/c784bb8bb6c1773fee160b9fdf7073670d1101d1

Video will be posted shortly.

**Comment by daveshap** [comment]
Here's the video recording of the process today. 

https://youtu.be/zzaEBGOVKIg

**Reply by russellballestrini** [reply]
Watching now.

**Comment by sirinath** [comment]
What would make me excited would be if:
1. You can build the whole knowledge base from 1st principles
2. Formally prove or disprove the solutions

**Reply by daveshap** [reply]
that will be required for constructing the dataset

**Reply by sirinath** [reply]
This this is a new project, take the opportunity to rethink everything. I am not an expert in this area but would be user.

If the models has 2 parts:
- memory
- inference

Any inference will update memory and knowledge will update the memory. Make it real time which invers the data presented and updates the memory. Inference validates the knowledge for validity. It can remember what was inferred and what was read. The inference can quickly update the memory without heavy training which can update real time. This way there is no cutoff. Memory serves as a catch for faster look up than inference. For answers which are which are purely inferential, the inference should be able to do it, a bit slower than recalling. 

The fact that there is training and also large data sets involved in current models is a very limiting factor in being able to live update and update oneself from the questions thrown at it.

The fact you have to train a model with large sets of data it self implies it is a glorified pattern recognition software. Also the inference part is patten recognition on top or pattern recognition, than logical inference and deduction.
